docs/user-guide/topics/ab_testing.md

```diff
@@ -0,0 +1,203 @@
+		// return
+	}
+	c := pb.NewABNClient(conn)
+	client = &c
+    ```
+
+### Define routing
+
+In the sample, track identifies are mapped to a static set of endpoints. The sample frontends maintain this in a simple map.
+
+=== "Node.js"
+    ```javascript
+    const trackToRoute = {
+        "backend":   "http://backend.default.svc.cluster.local:8091",
+        "backend-candidate-1": "http://backend-candidate-1.default.svc.cluster.local:8091",
+    }
+    ```
+
+=== "Go"
+    ```go
+    	trackToRoute = map[string]string{
+		"backend":             "http://backend.default.svc.cluster.local:8091",
+		"backend-candidate-1": "http://backend-candidate-1.default.svc.cluster.local:8091",
+	}
+    ```
+
+### Using *Lookup()*
+
+Given a user session identifier, *Lookup()* returns a track identifier that can be used to route requests. 
+
+In this example, the user session identifier is assumed to be passed in the `X-User` header of user requests.
+
+The track identifier is used as an index to the `trackToRoute` map defined above. A default is used if the call to *Lookup* fails for any reason.
+
+=== "Node.js"
+    ```javascript
+    var application = new messages.Application();
+    application.setName('default/backend');
+    application.setUser(req.header('X-User'));
+    client.lookup(application, function(err, session) {
+        if (err || (session.getTrack() == '')) {
+            // use default route (see above)
+            console.warn("error or null")
+        } else {
+            // use route determined by recommended track
+            console.info('lookup suggested track %s', session.getTrack())
+            route = trackToRoute[session.getTrack()];
+        }
+
+        // call backend service using route
+        ...
+    });
+    ```
+
+=== "Go"
+    ```go
+    route := trackToRoute["backend"]
+	user := req.Header["X-User"][0]
+    s, err := (*client).Lookup(
+		ctx,
+		&pb.Application{
+			Name: "default/backend",
+			User: user,
+		},
+	)
+    if err == nil && s != nil {
+		r, ok := trackToRoute[s.GetTrack()]
+		if ok {
+			route = r
+		}
+	}
+
+    // call backend service using route
+	...
+    ```
+
+### Using *WriteMetric()*
+
+In the samples, a single metric named *sample_metric* is assigned a random value between 0 and 100 and written to the metric store. In practice, the metric value should be meaningful.
+
+=== "Node.js"
+    ```javascript
+    // export metric
+    var mv = new messages.MetricValue();
+    mv.setName('sample_metric');
+    mv.setValue(random({min: 0, max: 100, integer: true}).toString());
+    mv.setApplication('default/backend');
+    mv.setUser(user);
+    ```
+
+=== "Go"
+    ```go
+    	_, _ = (*client).WriteMetric(
+		ctx,
+		&pb.MetricValue{
+			Name:        "sample_metric",
+			Value:       fmt.Sprintf("%f", rand.Float64()*100.0),
+			Application: "default/backend",
+			User:        user,
+		},
+	)
+    ```
+
+## Deployment Time Configuration of Backends
```

Comment:
Could we please push this up before the frontend configuration? It is easier to describe, so we may want to push this up first. Also, it ties in cleanly with the A/B/n service configuration description.